ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/lastresource/LastResourceAllowedTestCase.java

```diff
@@ -49,4 +108,103 @@ public void testAllowed()
             tm.rollback() ;
         }
     }
+
+    @Test
+    public void testHeuristicMixed() throws SystemException, NotSupportedException, RollbackException
+    {
+        // first last resource (LR) commits, second LR throws XA_HEURRB
+        // then validate that the commit produces a HeuristicMixedException
+        final LastOnePhaseResource firstResource = new LastOnePhaseResource(XAException.XA_HEURRB, 0);
+        final LastOnePhaseResource secondResource = new LastOnePhaseResource(0, 0);
+
+        // since both last resources will fail the expectation is rollback
+        generateLastResourceFailures(true, HeuristicMixedException.class.getName(), xar, firstResource, secondResource);
+    }
+
+    @Test
+    public void testRollback() throws SystemException, NotSupportedException, RollbackException
+    {
+        // both last resources throw XA_HEURRB
+        // then validate that the commit produces a RollbackException
+        final LastOnePhaseResource firstResource = new LastOnePhaseResource(XAException.XA_HEURRB, 0);
+        final LastOnePhaseResource secondResource = new LastOnePhaseResource(XAException.XA_HEURRB, 0);
+
+        // the prepare phase should fail on the first resource throws XA_HEURRB so the decision will switch to
+        // rolling back the remaining resources and the overall outcome should be a RollbackException
+        generateLastResourceFailures(true, RollbackException.class.getName(), xar, firstResource, secondResource);
+    }
+
+    @Test
+    @Ignore // ignored for now because:
```

Comment:
@mmusgrov  Is there plan (e.g.  a JBTM) to include this test (i.e. remove the ignore annotation)?